hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -495,17 +499,27 @@ protected boolean containsKeyInternal(Data key) {
         }
     }
 
+
+
     public void waitUntilLoaded() {
         try {
             int mapNamePartition = partitionService.getPartitionId(name);
+            if (Thread.currentThread() instanceof PartitionOperationThread) {
```

Comment:
isn't it user thread always? Is there a possibility to call `waitUntilLoaded ` from partition thread?